File: server/src/main/java/com/zhixun/common/result/ErrorCode.java
package com.zhixun.common.result;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 错误码枚举
 */
@Getter
@AllArgsConstructor
public enum ErrorCode {

    // ========== 通用错误码 2xx ==========
    SUCCESS(200, "success"),
    CREATED(201, "资源创建成功"),

    // ========== 客户端错误码 4xx ==========
    BAD_REQUEST(400, "请求参数错误"),
    PARAMS_ERROR(4001, "参数错误"),
    UNAUTHORIZED(401, "请先登录"),
    FORBIDDEN(403, "暂无操作权限"),
    NOT_FOUND(404, "请求的资源不存在"),
    METHOD_NOT_ALLOWED(405, "不支持的请求方式"),
    CONFLICT(409, "资源冲突，请稍后重试"),
    GONE(410, "资源已被删除"),
    UNPROCESSABLE_ENTITY(422, "提交信息不完整，请检查后重试"),
    TOO_MANY_REQUESTS(429, "操作太频繁，请稍后重试"),

    // ========== 服务端错误码 5xx ==========
    SERVICE_UNAVAILABLE(503, "服务正在维护中"),

    // ========== 认证相关 1xxx ==========
    AUTH_LOGIN_FAILED(1001, "登录失败，请检查账号密码"),
    AUTH_ACCOUNT_DISABLED(1002, "账号已被限制使用"),
    AUTH_TOKEN_EXPIRED(1003, "登录信息已过期，请重新登录"),
    AUTH_TOKEN_INVALID(1004, "登录信息已过期，请重新登录"),
    AUTH_REFRESH_TOKEN_EXPIRED(1005, "登录信息已过期，请重新登录"),
    AUTH_PHONE_NOT_VERIFIED(1006, "手机号码未验证"),
    AUTH_CAPTCHA_ERROR(1007, "验证码输入错误"),

    // ========== 用户相关 2xxx ==========
    USER_NOT_FOUND(2001, "用户不存在"),
    USER_PHONE_EXISTS(2002, "手机号已注册"),
    USER_EMAIL_EXISTS(2003, "邮箱已注册"),
    USER_PASSWORD_ERROR(2004, "密码输入错误"),
    USER_OLD_PASSWORD_ERROR(2005, "原密码输入错误"),

    // ========== 工单相关 3xxx ==========
    WORK_ORDER_NOT_FOUND(3001, "工单不存在"),
    WORK_ORDER_STATUS_ERROR(3002, "工单状态不正确"),
    WORK_ORDER_ASSIGNED(3003, "工单已分配"),
    WORK_ORDER_CLOSED(3004, "工单已关闭"),

    // ========== 文件相关 4xxx ==========
    FILE_UPLOAD_FAILED(4001, "文件上传失败，请重试"),
    FILE_TYPE_NOT_ALLOWED(4002, "不支持该文件格式"),
    FILE_SIZE_EXCEEDED(4003, "文件大小超出限制"),

    // ========== 业务错误 5xxx ==========
    SENSITIVE_WORD_DETECTED(5001, "内容包含不合规信息，请修改后重试"),
    OPERATION_TOO_FREQUENT(5002, "操作太频繁，请稍后重试"),
    BUSINESS_ERROR(5003, "操作失败，请稍后重试");

    /** 状态码 */
    private final int code;

    /** 提示信息 */
    private final String message;
}

File: packages/shared-types/error-code.ts
/**
 * 错误码枚举 — 前后端统一
 *
 * 分类规则：
 * - 通用：HTTP 标准状态码
 * - 1xxx：认证相关
 * - 2xxx：作品相关
 * - 3xxx：用户相关
 * - 4xxx：文件相关
 * - 5xxx：搜索相关
 */
export enum ErrorCode {
  // 通用
  SUCCESS = 0,
  BAD_REQUEST = 400,
  UNAUTHORIZED = 401,
  FORBIDDEN = 403,
  NOT_FOUND = 404,
  RATE_LIMIT = 429,
  INTERNAL_ERROR = 500,
  SERVICE_UNAVAILABLE = 503,

  // 认证 (1xxx)
  AUTH_USERNAME_OR_PASSWORD_ERROR = 1001,
  AUTH_TOKEN_EXPIRED = 1002,
  AUTH_TOKEN_INVALID = 1003,
  AUTH_REFRESH_TOKEN_INVALID = 1004,
  AUTH_ACCOUNT_BANNED = 1005,
  AUTH_ACCOUNT_NOT_FOUND = 1006,
  AUTH_VERIFICATION_CODE_ERROR = 1007,

  // 作品 (2xxx)
  ARTICLE_NOT_FOUND = 2001,
  ARTICLE_DELETED = 2002,
  ARTICLE_UNDER_REVIEW = 2003,
  ARTICLE_REJECTED = 2004,
  ARTICLE_FORBIDDEN_WORD = 2005,

  // 用户 (3xxx)
  USER_NOT_FOUND = 3001,
  USER_ALREADY_EXISTS = 3002,
  USER_BANNED = 3003,
  USER_FOLLOW_SELF = 3004,

  // 文件 (4xxx)
  FILE_TOO_LARGE = 4001,
  FILE_TYPE_NOT_ALLOWED = 4002,
  FILE_UPLOAD_FAILED = 4003,

  // 搜索 (5xxx)
  SEARCH_SERVICE_UNAVAILABLE = 5001,
  SEARCH_TOO_MANY_REQUESTS = 5002,
}

/** 错误码对应的中文提示信息 */
export const ErrorMessage: Record<number, string> = {
  [ErrorCode.SUCCESS]: '操作成功',
  [ErrorCode.BAD_REQUEST]: '请求参数错误',
  [ErrorCode.UNAUTHORIZED]: '请先登录',
  [ErrorCode.FORBIDDEN]: '没有权限',
  [ErrorCode.NOT_FOUND]: '内容不存在',
  [ErrorCode.RATE_LIMIT]: '操作太频繁，稍后再试',
  [ErrorCode.INTERNAL_ERROR]: '服务器开小差了',
  [ErrorCode.SERVICE_UNAVAILABLE]: '服务正在维护中',
  [ErrorCode.AUTH_USERNAME_OR_PASSWORD_ERROR]: '登录失败，请检查账号密码',
  [ErrorCode.AUTH_TOKEN_EXPIRED]: '登录信息已过期，请重新登录',
  [ErrorCode.AUTH_TOKEN_INVALID]: '登录信息已过期，请重新登录',
  [ErrorCode.AUTH_REFRESH_TOKEN_INVALID]: '登录信息已过期，请重新登录',
  [ErrorCode.AUTH_ACCOUNT_BANNED]: '账号已被限制使用',
  [ErrorCode.AUTH_ACCOUNT_NOT_FOUND]: '账号不存在',
  [ErrorCode.AUTH_VERIFICATION_CODE_ERROR]: '验证码输入错误',
  [ErrorCode.ARTICLE_NOT_FOUND]: '作品不存在或已被删除',
  [ErrorCode.ARTICLE_DELETED]: '作品已被作者删除',
  [ErrorCode.ARTICLE_UNDER_REVIEW]: '作品正在审核中，请耐心等待',
  [ErrorCode.ARTICLE_REJECTED]: '作品未通过审核',
  [ErrorCode.ARTICLE_FORBIDDEN_WORD]: '作品包含不合规内容，请修改后重试',
  [ErrorCode.USER_NOT_FOUND]: '该用户不存在',
  [ErrorCode.USER_ALREADY_EXISTS]: '该账号已被注册',
  [ErrorCode.USER_BANNED]: '该用户已被限制使用',
  [ErrorCode.USER_FOLLOW_SELF]: '不能关注自己哦',
  [ErrorCode.FILE_TOO_LARGE]: '文件大小超出限制',
  [ErrorCode.FILE_TYPE_NOT_ALLOWED]: '不支持该文件格式',
  [ErrorCode.FILE_UPLOAD_FAILED]: '文件上传失败，请重试',
  [ErrorCode.SEARCH_SERVICE_UNAVAILABLE]: '搜索服务正在维护中',
  [ErrorCode.SEARCH_TOO_MANY_REQUESTS]: '搜索太频繁，请稍后重试',
}
